superset/assets/javascripts/explore/propTypes/savedMetricType.js

```diff
@@ -0,0 +1,6 @@
+import PropTypes from 'prop-types';
+
+export default PropTypes.shape({
+  metric_name: PropTypes.string.isRequired,
+  expression: PropTypes.string.isRequired,
+});
```

Comment:
Just curious, why have this in a separate file instead of defined as a shape in the file it's used? This isn't being used in multiple places and it doesn't have a complicated structure (with imports). Is it a best practice?

Reply:
Hmm yeah I could move it into MetricsControl- I agree this doesn't help much.